rolling-update-kubernetes/src/main/scala/akka/rollingupdate/kubernetes/AppVersionRevision.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2017-2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.rollingupdate.kubernetes
+
+import akka.util.Version
+import akka.actor.ActorSystem
+import akka.actor.ExtendedActorSystem
+import akka.actor.Extension
+import akka.actor.ExtensionId
+import akka.actor.ExtensionIdProvider
+import akka.cluster.Cluster
+import akka.event.Logging
+
+import java.util.concurrent.atomic.AtomicBoolean
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.concurrent.Promise
+import scala.util.control.NonFatal
+
+final class AppVersionRevision(implicit system: ExtendedActorSystem) extends Extension {
+
+  private val log = Logging(system, classOf[AppVersionRevision])
+  private val configPath = "akka.rollingupdate.kubernetes"
+  private val config = system.settings.config.getConfig(configPath)
+  private val k8sSettings = KubernetesSettings(config)
+  implicit private val ec: ExecutionContext = system.dispatcher
+  private final val isInitialized = new AtomicBoolean(false)
+  log.debug("Settings {}", k8sSettings)
+
+  private val versionPromise = Promise[Version]()
+
+  def getRevision(): Future[Version] = versionPromise.future
+
+  def start(): Unit = {
+    if (k8sSettings.podName.isEmpty) {
+      log.warning(
+        "No configuration found to extract the pod name from. " +
+        s"Be sure to provide the pod name with `$configPath.pod-name` " +
+        "or by setting ENV variable `KUBERNETES_POD_NAME`.")
```

Comment:
Something about the consequence of it missing? "The app version will not be picked up from Kubernetes" or something like that? It may not be crash-critical but maybe it should even be an error log as it clearly is something wrong when the extension is enabled?